### PR TITLE
fix(components): fixes automatic uncovering of cells

### LINKED ISSUE #49: Cell updates are not rendered
After modifying the state of cells (which are references), the UI does not get updated.
It seems that some unit tests for that are missing.

### LINKED ISSUE #51: Uncovering a cell with no adjacent mines tries to uncover flagged adjacent cells
When uncovering a `"0"`, we "flood fill" every adjacent `"0"` as well for convenience.
If cells, which are about to be "flood filled", have been flagged, the "flood fill" tries to uncover those flagged cells, which will result in `InvalidOperationException: "Uncover is not allowed from the status 'Flagged'."`

### PR DESCRIPTION
**Pull request by dev-in-disguise**
Cells which are uncovered automatically due to the 0 adjacent mines feature are now re-rendered correctly.
Cells in the status covered and flagger which are tried to be uncovered automatically due to the 0 adjacent mines feature are no longer tried to be uncovered in the ui.

fixes #49 , fixes #51

### CODE REVIEW

File: source/production/F0.Minesweeper.Components/Logic/Cell/DefaultCellStatusManager.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ private DefaultCellStatusManager()
 
 		internal static DefaultCellStatusManager Instance => lazyInstance.Value;
```
[the remark this answers is not in the document]
Comment (Flash0ver) [reply source/production/F0.Minesweeper.Components/Logic/Cell/DefaultCellStatusManager.cs:15]: But just realized we use it at more than just one position ... perhaps we could wrap the nullable `ICellStatusManager` in an non-nullable property which does the `Debug.Assert(statusManager is not null)` ... but not too sure about that ...

File: source/production/F0.Minesweeper.App/Properties/launchSettings.json [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
-			"dotnetRunMessages": "true",
-			"launchBrowser": true,
-			"inspectUri": "{wsProtocol}://{url.hostname}:{url.port}/_framework/debug/ws-proxy?browser={browserInspectUri}",
-			"applicationUrl": "http://localhost:5000",
-			"environmentVariables": {
-				"ASPNETCORE_ENVIRONMENT": "Development"
-			}
-		}
-	}
+  "iisSettings": {
```
[the remark this answers is not in the document]
Comment (dev-in-disguise) [reply source/production/F0.Minesweeper.App/Properties/launchSettings.json:2]: Quite funny that actually space takes up way more ... well ... _space_ ... than tabs in the real world.
fixed :)

File: source/production/F0.Minesweeper.Components/Minefield.razor.cs [hunk 3]
```diff
@@ -47,23 +48,31 @@ protected override void OnAfterRender(bool firstRender)
...
 			}
 
-			IGameUpdateReport report = minefield.Uncover(location);
+			IGameUpdateReport report = minefield.Uncover(clickedLocation);
 
-			foreach(Cell cell in cells)
+
+			var uncoverableCells =
+				(
+					from cell in cells
```
[the remark this answers is not in the document]
Comment (dev-in-disguise) [reply source/production/F0.Minesweeper.Components/Minefield.razor.cs:63]: Me too. Was quite a pleasure using these rusty skills once again 😄

File: source/production/F0.Minesweeper.Components/Logic/Cell/DefaultCellStatusManager.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (dev-in-disguise) [reply source/production/F0.Minesweeper.Components/Logic/Cell/DefaultCellStatusManager.cs:15]: I tried to come up with a good solution but I just can't. The three possibilities we currently have:

1. continue with the `DefaultStatusManager`
2. use `Debug.Assert`
3. use the _null-forgiving operator_
 
And to be honest I like none of them. They all just try to brush over an issue that will be hopefully fixed with [this feature in .net 6](https://github.com/orgs/dotnet/projects/20#card-53920288). 

There's one further possibility that came to my head: 
```
public Cell() : this(CellStatusManagerFactory.GetManager())
{
}

public Cell(ICellStatusManager statusManager)
{
	statusText = MapToText(CellStatusType.Covered);
	this.statusManager = statusManager;
}
```
But this is basically adding a new constructor just for tests. And I'm not too fond of adding code just for tests. What do you think about this?

**Comment by mcNicolescu** [comment]
Hi Manuel

My preferred solution for such scenarios is a small factory interface, that
can be injected and has a Create method
It serves both purposes without modifications to the code base
For testing purpose, we just inject the "testable" factory

It may lead to a few lines of code more than the ctor overload

KR





Marius Constantin Nicolescu
Senior Technology Architect
Tel:   +43-699-19 20 49 20

**Comment by mcNicolescu** [comment]
Even if we don't inject the factory, the important part is that being an
interface, the Create method of the factory can be mocked easily in a test.
;)

File: source/production/F0.Minesweeper.Components/Logic/Cell/DefaultCellStatusManager.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (dev-in-disguise) [reply source/production/F0.Minesweeper.Components/Logic/Cell/DefaultCellStatusManager.cs:15]: We decided to wait for the .net 6 functionality

**Review by Flash0ver — APPROVED** [review]

**Review by thomas-pemsel — APPROVED** [review]

**Pull request merged**